Databases/SQL/Tips & Tricks.md

```diff
@@ -0,0 +1,47 @@
+## N-queries
+
+You should avoid multiple DB queries from the application if you can fetch needed data in a single query. Always check what queries are executed in your (Rails) log.
+
+Also, when you work on a complex SQL calculation, avoid pulling intermediate results to your application, which will be used for the next step in your query. It can be a very resource-intensive operation.
+
+
+## UNION (ALL)
+
+When you use `UNION` operator for combining result sets of two or more query statements, double-check if you would use `UNION` or `UNION ALL`.
+
+The `UNION` operator is *less performant* because it **removes all duplicate rows** from the combined data set.
+To retain the duplicate rows, you use the `UNION ALL` instead.
+
+
+## VIEWs
```

Comment:
I'd maybe mention few more things here:

- you can use a view as an ActiveRecord model (with example)
- scenic gem for managing views?
- are materialized views a way to go if you have a lot of inserts (you'd need to refresh it constantly)
- tips on when to use which view?